x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchBaseRestHandler.java

```diff
@@ -28,6 +28,9 @@ protected EnterpriseSearchBaseRestHandler(XPackLicenseState licenseState, Licens
     protected final BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         if (LicenseUtils.supportedLicense(this.licenseState)) {
             return innerPrepareRequest(request, client);
+        } else if (this.product.equals(LicenseUtils.Product.CONNECTOR)) {
```

Comment:
Rather than adding this check, why not have the connectors handlers just implement `BaseRestHandler`?